storage/snapshot/snapshot.rs

```diff
@@ -409,6 +425,7 @@ impl Error for SnapshotError {
 #[derive(Debug, Clone)]
 pub enum SnapshotGetError {
     MVCCRead { source: MVCCReadError },
+    GetRequiredKeyDoesntExist { key: StorageKey<'static, BUFFER_KEY_INLINE> }
```

Comment:
Do we need to return the key?

Reply:
For something that's an internal error I propose we call them `Expected<Precondition>`, in this case `ExpectedRequiredKeyToExist`.
Returning the key seems helpful.